azext_edge/edge/providers/support/base.py

```diff
@@ -504,11 +450,54 @@ def _capture_pod_container_logs(
                         "data": log,
                         "zinfo": (
                             f"{pod_namespace}/{resource_api.moniker}"
-                            f"/pod.{pod_name}.{container.name}.{zinfo_previous_segment}log"
+                            f"/{child_group}pod.{pod_name}.{container.name}.{zinfo_previous_segment}log"
                         ),
                     }
                 )
             except ApiException as e:
                 logger.debug(e.body)
 
     return processed
+
+
+def _process_kubernetes_resources(
+    resources: object,
+    resource_api: EdgeResourceApi,
+    child_group: Optional[str] = None,
+    prefix_names: Optional[List[str]] = None,
+    kind: Optional[str] = None,
+) -> List[dict]:
+    processed = []
+
+    if not prefix_names:
+        prefix_names = []
+
+    logger.info(f"Detected {len(resources.items)} {kind}s.")
+    for resource in resources.items:
+        r = resource
+        r.api_version = resources.api_version
+        r.kind = kind
+        resource_metadata = r.metadata
+        resource_namespace = resource_metadata.namespace
+        resource_name = resource_metadata.name
+
+        if prefix_names:
+            matched_prefix = [resource_name.startswith(prefix) for prefix in prefix_names]
+            if not any(matched_prefix):
+                continue
+
+        if len(kind) > 12:
+            # get every first capital letter in the kind
+            resource_type = "".join([c for c in kind if c.isupper()]).lower()
+        else:
+            resource_type = kind.lower()
+
+        processed.append(
+            {
+                "data": generic.sanitize_for_serialization(obj=r),
+                "zinfo": f"{resource_namespace}/{resource_api.moniker}/"
+                f"{child_group}{resource_type}.{resource_name}.yaml",
```

Comment:
child_group is typed as Optional[str] = None ,  if nothing is passed for child_group this segment of the string becomes "None{resource_type}.{resource_name}.yaml"